### PR TITLE
SWARM-817: wildfly-swarm:run does not work for dependencies added with DependenciesContainer.addAllDependencies()

### PR DESCRIPTION
**Pull request by heiko-braun**
- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

- Shared dependency information between maven plugin and process started from process builder (i.e swarm:run)
- Use a bucket based approach to declared dependencies (easier to reason about the origin of a transient dep)

### CODE REVIEW

**Comment by heiko-braun** [comment]
DON'T MERGE: This is just to keep CI busy while I do other things...

**Comment by kenfinnigan** [comment]
is this ok to review or not yet?

**Comment by heiko-braun** [comment]
Not yet. On monday.

**Comment by heiko-braun** [comment]
Remaining TODO before this can be reviewed:

- [x] Consolidate DependencyTree usage across modules
- [x] Fix the gradle plugin

**Comment by heiko-braun** [comment]
Left todo:
- [x] Verify the examples

File: tools/src/main/java/org/wildfly/swarm/tools/BuildTool.java [hunk 1]
```diff
@@ -329,7 +329,7 @@ private void addFractions(ResolvedDependencies resolvedDependencies) throws Exce
                                       .sorted()
                                       .collect(Collectors.toList())));
 
-        allFractions.forEach(f -> this.declaredDependencies.addExplicitDependency(f));
+        allFractions.forEach(f -> this.declaredDependencies.add(f)); // TODO
```
Comment (kenfinnigan) [inline tools/src/main/java/org/wildfly/swarm/tools/BuildTool.java]: What is it that needs to be done here?

**Review by kenfinnigan — APPROVED** [review]

File: tools/src/main/java/org/wildfly/swarm/tools/BuildTool.java (hunk 1 above)
Comment (heiko-braun) [reply tools/src/main/java/org/wildfly/swarm/tools/BuildTool.java]: that's a stale comment. i've removed it.

**Pull request merged**